### PR TITLE
Add remand_reason ODS table for VHA

### LINKED ISSUE #13733: VHA ODS tables
## Description

VHA has reporting needs similar to those of VBA. These include tables:

* HIGHER_LEVEL_REVIEWS: Intake data for Higher level Reviews
* SUPPLEMENTAL CLAIMS: Intake data for Supplemental Claims
* APPEALS: Used to keep track of information for AMA appeals
* DECISION_ISSUES: Issue-level dispositions for AMA claims/appeals
* REMAND_REASONS: Remand reason for decision issues.

In order to make it simple for VHA to pull the data they need, we need a new ODS table `vha_decision_reviews` that is scoped to those Decision Reviews (Appeal, HLR, SC) that have an associated benefit type corresponding to VHA.

## Acceptance criteria

* [x] New `vha_decision_reviews` ODS table with composite schema matching HLR, SC and Appeal.
* [ ] Add `remand_reasons` ODS table
* [x] `ETL::VHADecisionReview` model and syncer added to builder job

## Technical notes

Easiest way to identify VHA-related decision reviews is via the BusinessLine to task association. See #13075 for example.

- [x] Verify that nonrating lines of business that map to VHA are included (does benefit type always == LoB) verified via https://github.com/department-of-veterans-affairs/dsva-vacols/issues/77

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves #13733
Follow-on PR to first PR #14208.

### Description
Add ETL remand_reason table for VHA reporting needs. 
Pretty much a straight copy of original table.

### Acceptance Criteria
- [x] Code compiles correctly

### Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Database Changes
*Only for Schema Changes*

* [x] Timestamps (created_at, updated_at) for new tables
* [x] Column comments updated
* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [x] DB schema docs updated with `make docs`
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

**Comment by yoomlam** [comment]
For reference once this PR is merged:
```
irb(main):001:0> RemandReason.count
[2020-05-17 10:42:22 -0400]    (5.7ms)  SELECT COUNT(*) FROM "remand_reasons"
=> 27296
```

File: app/models/etl/remand_reason.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# Copy of remand_reasons table
+
+class ETL::RemandReason < ETL::Record
+  class << self
+    private
+
+    def merge_original_attributes_to_target(original, target)
+      target.attributes = original.attributes.reject { |key| %w[created_at updated_at].include?(key) }
```
Comment (pkarman) [inline app/models/etl/remand_reason.rb:10]: agreed with this reuse of the PK `id`

**Comment by yoomlam** [comment]
Manually ran syncer since I forgot to include it in `ETLClasses`:
```
etl_build=ETL::Build.create
sync=ETL::RemandReasonSyncer.new(etl_build: etl_build)
sync.call

RemandReason.count
=> 27756

ETL::RemandReason.count
=> 27756
```
It works!

**Comment by yoomlam** [comment]
Looks like it's updating:
```ruby
ETL::RemandReason.count
=> 28057
RemandReason.count
=> 28098
```

**Pull request merged**